iip-5.md

```diff
@@ -0,0 +1,52 @@
+```
+IIP: 5
+Title: Classifying EVM Error Status
+Author: Dorothy Ko (dorothy@iotex.io)
+Status: Draft
+Type: Standards Track
+Created: 2019-08-14
+```
+
+
+## Abstract
+Currently every execution of smart contract on IoTex Blockchain doesn't show any information of error if it fails to execute or deploy the smart contract. In order to give an information of error to the end-user, this standard outlines a common set of EVM Error Status Code. Therefore, we classify the possible EVM error and define EVM error status code. And we propose to give the EVM Error Code to the Receipt Status, so that it is visible to users/developers and accordingly it enhances UX(User Experience) and DX(Developer Experience). 
+
+## Motivation
+Sometimes, we failed to execute or deploy a smart contract. However, currently its status gives us just simple status such as Success(1) or Failure(0) in receipt and it doesn't give us any information of the reason of failure. In terms of user and developer experience, it makes us difficult to utilize the smart contract execution.  
+
+## Specification
+
+> General Status 
+
+| code | Description  |
+|------|--------------|
+| 0    | Failure      |
+| 1    | Success      |
+
+
+> 1xx for EVM Execution Error Status 
+
+| code | 						Description                               |
+|------|------------------------------------------------------------------|
+| 100  | ErrUnknown : Unknown                                             |
+| 101  | ErrOutOfGas : Out of Gas                                         |
+| 102  | ErrCodeStoreOutOfGas : Contract creation code storage out of gas |
```

Comment:
what is diff between 101 and 102?

Reply:
Actually this error code is from EVM (geth), therefore according to the geth code, ErrOutofGas(101) is general out of gas error when we don't have enough gas to run the contract, but ErrCodeStoreOutOfGas(102) happens especially because of not enough gas to store the contract code when we deploy the contract.